x-pack/legacy/plugins/siem/server/lib/framework/kibana_framework_adapter.ts

```diff
@@ -21,125 +26,113 @@ import {
   WrappableRequest,
 } from './types';
 
-interface CallWithRequestParams extends GenericParams {
-  max_concurrent_shard_requests?: number;
-}
-
 export class KibanaBackendFrameworkAdapter implements FrameworkAdapter {
   public version: string;
-  public envMode: EnvironmentMode;
+  private isProductionMode: boolean;
+  private router: IRouter;
 
-  constructor(private server: ServerFacade, mode: EnvironmentMode) {
-    this.version = server.config().get('pkg.version');
-    this.envMode = mode;
+  constructor(core: CoreSetup, __legacy: ServerFacade) {
+    this.version = __legacy.config().get('pkg.version');
+    this.isProductionMode = process.env.NODE_ENV === 'production';
+    this.router = core.http.createRouter();
   }
 
   public async callWithRequest(
     req: FrameworkRequest,
     endpoint: string,
-    params: CallWithRequestParams,
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    ...rest: any[]
+    params: Record<string, any>
   ) {
-    const internalRequest = req[internalFrameworkRequest];
-    const { elasticsearch } = internalRequest.server.plugins;
-    const { callWithRequest } = elasticsearch.getCluster('data');
-    const includeFrozen = await internalRequest.getUiSettingsService().get('search:includeFrozen');
+    const { elasticsearch, uiSettings } = req.context.core;
+    const includeFrozen = await uiSettings.client.get('search:includeFrozen');
     const maxConcurrentShardRequests =
       endpoint === 'msearch'
-        ? await internalRequest.getUiSettingsService().get('courier:maxConcurrentShardRequests')
+        ? await uiSettings.client.get('courier:maxConcurrentShardRequests')
         : 0;
-    const fields = await callWithRequest(
-      internalRequest,
-      endpoint,
-      {
-        ...params,
-        ignore_throttled: !includeFrozen,
-        ...(maxConcurrentShardRequests > 0
-          ? { max_concurrent_shard_requests: maxConcurrentShardRequests }
-          : {}),
-      },
-      ...rest
-    );
-    return fields;
-  }
 
-  public exposeStaticDir(urlPath: string, dir: string): void {
-    this.server.route({
-      handler: {
-        directory: {
-          path: dir,
-        },
-      },
-      method: 'GET',
-      path: urlPath,
+    return elasticsearch.dataClient.callAsCurrentUser(endpoint, {
+      ...params,
+      ignore_throttled: !includeFrozen,
+      ...(maxConcurrentShardRequests > 0
+        ? { max_concurrent_shard_requests: maxConcurrentShardRequests }
+        : {}),
     });
   }
 
   public registerGraphQLEndpoint(routePath: string, schema: GraphQLSchema): void {
-    this.server.route({
-      options: {
-        tags: ['access:siem'],
+    this.router.get(
```

Comment:
I couldn't actually find any places that we were GETing these graphQL endpoints; all I saw were POSTs. @XavierM do you know whether this GET route is used?

Reply:
Followup: this endpoint is only used by graphiQL. Moving this route declaration to the same branch as the graphiQL endpoint for clarity.